record/hop.go

```diff
@@ -43,5 +43,10 @@ func NewLockTimeRecord(lockTime *uint32) tlv.Record {
 // NewNextHopIDRecord creates a tlv.Record that encodes the short_channel_id
 // (type 6) for an onion payload.
 func NewNextHopIDRecord(cid *uint64) tlv.Record {
-	return tlv.MakePrimitiveRecord(NextHopOnionType, cid)
+	return tlv.MakeDynamicRecord(
+		NextHopOnionType, cid, func() uint64 {
+			return tlv.SizeTUint64(*cid)
+		},
+		tlv.ETUint64, tlv.DTUint64,
+	)
```

Comment:
Is a more elaborate commit message needed, as this commit is (in theory) partitioning the network?

Reply:
now that we simply omit the field, rather than change the encoding it's no longer a partition since it wasn't being enforced before. this code has never been deployed in a release so i think its less critical even if other impls are enforcing it

Reply:
Yeah this doesn't change any behavior, as if the cid wasn't present, then we would be passing around the default value, which in this case, is the same as `exitHop`.